api/tools/src/main/java/org/pmiops/workbench/tools/DeleteWorkspaces.java

```diff
@@ -0,0 +1,175 @@
+package org.pmiops.workbench.tools;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.time.Clock;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Provider;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.dao.UserRecentWorkspaceDao;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.DbStorageEnums;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.firecloud.ApiClient;
+import org.pmiops.workbench.firecloud.FireCloudConfig;
+import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.firecloud.FireCloudServiceImpl;
+import org.pmiops.workbench.firecloud.api.ProfileApi;
+import org.pmiops.workbench.firecloud.api.WorkspacesApi;
+import org.pmiops.workbench.model.WorkspaceActiveStatus;
+import org.pmiops.workbench.workspaces.WorkspaceService;
+import org.pmiops.workbench.workspaces.WorkspaceServiceImpl;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Scope;
+
+@Configuration
+@Import({FireCloudServiceImpl.class, FireCloudConfig.class})
+public class DeleteWorkspaces {
+
+  private static final Logger log = Logger.getLogger(DeleteWorkspaces.class.getName());
+
+  private static Option deleteListFilename =
+      Option.builder()
+          .longOpt("delete-list-filename")
+          .desc(
+              "File containing list of workspaces to delete. Each line should contain a single workspace's namespace and firecloud name, separated by a comma"
+                  + "Example: ws-namespace-1,fc-id-1 \n ws-namespace-2,fc-id-2 \n ws-namespace-3, fc-id-3")
+          .required()
+          .hasArg()
+          .build();
+
+  private static Options options = new Options().addOption(deleteListFilename);
+
+  @Bean
```

Comment:
Was this easier than simply implementing a batch delete API endpoint? I'd rather have swagger boilerplate than have to hack the service injection like this.

Reply:
For me, yes. Adding it as an admin endpoint would involve adding a one-off to our API, adding a new authority role, and writing some front end to fire the endpoint. Writing the calls in ruby/node isn't trivial either since we don't have API client packages for either and we'd have to figure out an authentication workflow to fire them.

Reply:
I know it feels heavy, but some of these are one-time things that will pay off.

For example, if we don't already have an administrative/devops authority, that's something I'll be wanting. If you want to use Ruby, I've already got a ticket to start generating a client library for that at https://precisionmedicineinitiative.atlassian.net/browse/RW-4499. But really, couldn't you just use `curl` for this?

The other consideration is that we already have testing patterns for this kind of thing, but with the tools, it's less clear how we make sure they keep working if they're infrequently used.